### PR TITLE
feat(PlaceholderContainer): add placeholder container component

### PR DESCRIPTION
**Pull request by Marginy605**

### CODE REVIEW

File: src/components/PlaceholderContainer/PlaceholderContainer.scss [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+$imageSmallSize: 100px;
+$imageMediumSize: 150px;
+$imageLargeSize: 230px;
+$imagePromoSize: 200px;
+
+$block: '.#{variables.$ns}placeholder-container';
+
+#{$block} {
+    $class: &;
```
Comment (amje) [inline src/components/PlaceholderContainer/PlaceholderContainer.scss:14]: $class is the same as $block, no need to have it

File: src/components/PlaceholderContainer/PlaceholderContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    title?: string;
+    description?: React.ReactNode;
+    html?: React.ReactNode;
+    action?: Action | Action[];
+    className?: string;
+    image: PlaceholderContainerImageNodeProps | PlaceholderContainerImageSettingsProps;
+}
+
+interface PlaceholderContainerDefaultProps {
+    entity: string;
```
Comment (amje) [inline src/components/PlaceholderContainer/PlaceholderContainer.tsx:43]: This shouldn't be there

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/PlaceholderContainer/PlaceholderContainer.scss (hunk 1 above)
Comment (Marginy605) [reply src/components/PlaceholderContainer/PlaceholderContainer.scss]: fixed

File: src/components/PlaceholderContainer/PlaceholderContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            );
+        }
+
+        return this.props.image;
+    }
+
+    private renderContent() {
+        const {html, contentSize, contentClassName} = this.props;
+
+        const content = html ? (
```
Comment (amje) [inline src/components/PlaceholderContainer/PlaceholderContainer.tsx:107]: Let's use `renderContent` function prop instead of `html`

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/PlaceholderContainer/PlaceholderContainer.scss [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            @include mixins.text-subheader-3();
+        }
+
+        #{$block}_size_promo & {
+            @include mixins.text-header-1();
+        }
+    }
+
+    &__description {
+        margin-top: 10px;
```
[the remark this answers is not in the document]
Comment (Marginy605) [reply src/components/PlaceholderContainer/PlaceholderContainer.scss]: fixed

File: src/components/PlaceholderContainer/PlaceholderContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+} as const;
+
+type Size = 's' | 'm' | 'l' | 'promo';
+
+type PlaceholderContainerImageNodeProps = NonNullable<React.ReactNode>;
+
+type PlaceholderContainerImageSettingsProps = {
+    url: string;
+    alt?: string;
+    size?: number;
```
Comment (amje) [inline src/components/PlaceholderContainer/PlaceholderContainer.tsx:32]: Not sure if we need this prop, we already have image constraints. In fact, image from url can be rectangle shape or smaller size than image contanier

File: src/components/PlaceholderContainer/PlaceholderContainerAction/PlaceholderContainerAction.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import {Button, ButtonSize, ButtonView} from '@gravity-ui/uikit';
+
+import {block} from '../../utils/cn';
+
+import './PlaceholderContainerAction.scss';
+
+const b = block('placeholder-container-action');
```
Comment (amje) [inline src/components/PlaceholderContainer/PlaceholderContainerAction/PlaceholderContainerAction.tsx:9]: This should be the same block `placeholder-container` and action is an element of the block

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/PlaceholderContainer/PlaceholderContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+            return null;
+        }
+
+        if (Array.isArray(action)) {
+            if (!action.length) {
+                return null;
+            }
+
+            return (
+                <div className={b('action')}>
```
[the remark this answers is not in the document]
Comment (Marginy605) [reply src/components/PlaceholderContainer/PlaceholderContainer.tsx:156]: fixed

File: src/components/PlaceholderContainer/PlaceholderContainer.tsx (hunk 5 above)
Comment (Marginy605) [reply src/components/PlaceholderContainer/PlaceholderContainer.tsx:32]: That's a good point. I removed this prop.

**Review by amje — APPROVED** [review]

**Pull request merged**